mandible/umm/ummg.py

```diff
@@ -0,0 +1,225 @@
+from dataclasses import asdict, dataclass
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import List
+
+# ISO 8601 with Z
+UMM_DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+
+def _remove_missing(ummg: dict) -> dict:
+    return {k: v for k, v in ummg.items() if v}
+
+
+@dataclass
+class FileMd(dict):
+    name: str
+    uri: str
+    s3uri: str
+    size: str
+    key: str
+    bucket: str
+
+
+@dataclass
+class CollectionRef:
+    ShortName: str
+    LongName: str
+
+
+@dataclass
+class ProductMd:
+    productCreationDateTime: datetime
+    productStartTime: datetime
+    productStopTime: datetime
+    mission: str
+
+
+@dataclass
+class ProductFilesMd:
+    productFile: FileMd
+    relatedFiles: List[FileMd]
+
+
+@dataclass
+class Meta:
+    CollectionRef: CollectionRef
+    ProductMd: ProductMd
+    ProductFilesMd: ProductFilesMd
+
+
+class UmmgBase():
+    def __init__(self, meta: Meta):
+        meta = asdict(meta)
+        self.now = datetime.now(timezone.utc)
+        self.set_product_metadata(meta)
+        self.set_product_files(meta)
+        self.set_collection_ref_table(meta)
+
+    # object setters
+    def set_collection_ref_table(self, meta: dict):
+        self.collection_ref_table = meta["CollectionRef"]
+
+    def set_product_metadata(self, meta: dict):
+        self.product_metadata = meta["ProductMd"]
+
+    def set_product_files(self, meta: dict):
+        self.product_files = meta["ProductFilesMd"]
+
+    # general getters used in the ummg getters
+    def get_file_name_path(self):
+        return Path(self.get_product_file_metadata()["name"])
+
+    def get_file_type(self) -> str:
+        file_type = {
+            ".bin": "Binary",
+            ".gpkg": "GeoPackage",
+            ".h5": "HDF5",
+            ".json": "JSON",
+            ".xml": "XML",
+        }
```

Comment:
This could still be a global constant

Reply:
I guess it would technically not be constant. It would be possible for a user of the library to import it and insert additional keys into it at runtime.

Reply:
I debated 2 options. I sort of wanted the user of the class to be able to extend it, but then I realized putting it in the init maybe could be annoying. I was sort of envisioning a user being able to append to that dict with the extra keys they need? So maybe putting it in the init?